Stress_testing.cpp

```diff
@@ -0,0 +1,55 @@
+/*Stress Testing
+It is done to check if our method is working fine for each and every test case, we generate some random values using rand
+function or even if we want to get arrray we can get by filling an array through loop and rand function.
+
+what we do in stress testing
+We use two functions one we know is working well and other function we want to check, we just give our values to these
+functions and check if both fubctions are giving the same output.
+*/
+#include <bits/stdc++.h>
+using namespace std;
+
+#define ll long long
+
+void fun(int a, int b) 
+	int c, d, s1 = 0, s2 = 0, s3 = 0, s4 = 0;
+	c = a;
+	d = b;
+	while(a != 0 || b != 0) {
+				s1 += a%10;
+				s2 += b%10;
+				a /= 10;
+				b /= 10;
+			}
+	a = c;
+	b = d;		
+	while(a != 0) {
+				s3 += a%10;
+				a /= 10;
+			}
+	while(b != 0) {
+				s4 += b%10;
+				b /= 10;
+			}
+	a = c;
+	b = d;
+	if(s1 != s3 || s2 != s4)
+	cout<<"Breaked for "<<a<<"  "<<b<<"\n";
+}

```

Comment:
Do you think this is the effective way of doing the stress testing in any algorithm

Reply:
This is not the actual way its just an example showing how Stress testing is done.
It was designed to check the robustness of a software by testing beyond test cases.

Reply:
chandreshji  and aksdeepji  please explain about it more .

Reply:
#include<iostream>
using namespace std;

void fun1(int *p, int *q) {
	int a = *p;
	int b = *q;
	int s1 = 0, s2 = 0;
	while(a != 0 || b != 0) {
				s1 += a%10;
				s2 += b%10;
				a /= 10;
				b /= 10;
			}
			*p = s1;
			*q = s2;
}
void fun2(int *p, int *q) {
		int a = *p;
		int b = *q;
		int s1 = 0, s2 = 0;
		while(a != 0) {
				s1 += a%10;
				a /= 10;
			}
		while(b != 0) {
				s2 += b%10;
				b /= 10;
			}
			*p = s1;
			*q = s2;
}
int main() {
	while(1) {
		int a, b, x, y;
		a = rand()%100000 + 1;
		b = rand()%100000 + 1;
		x = a;
		y = b;
		cout<<"Case "<<a<<"  "<<b<<"\t";
		fun1(&a, &b);
		fun2(&x, &y);
		cout<<"Sum is "<<a<<"  "<<b<<"\n";
		if(a != x || b != y) {
			cout<<"Breaked for "<<a<<"  "<<b;
			break;
		}
	}
	return 0;
}

Reply:
Hope you understand by this example i have used two functions performing the same task i.e. sum of digits of a number.
In case sum of digits returned by function 1 does not matches with function 2, the loop breaks and prints the case for which they are not working fine.